### PR TITLE
Intake FSM Implemented and Simulated

### PR DESCRIPTION
**Pull request by SiddharthKadari**
FSM Diagram -- https://docs.google.com/drawings/d/1r3vucw7IESH8VHF1JEroV67yvO6-PlUOcloL6tB5TLk/edit
Current FSM Diagram is fully implemented. System simulated, one issue with setting motor power.

### CODE REVIEW

File: src/main/java/frc/robot/systems/FSMSystem.java [hunk 1]
```diff
@@ -1,132 +1,283 @@
 package frc.robot.systems;
 
+import javax.lang.model.util.ElementScanner6;
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/FSMSystem.java]: Is this used for something?

File: src/main/java/frc/robot/TeleopInput.java [hunk 2]
```diff
@@ -84,6 +84,14 @@ public double getRightJoystickY() {
 		return rightJoystick.getY(GenericHID.Hand.kRight);
 	}
 
+	public boolean isRampToggleButtonPressed(){
+		return rightJoystick.getRawButton(1);
+	}
+
+	// public boolean isRetractButtonPressed(){
+	// 	return rightJoystick.getRawButton(2);
+	// }
```
Comment (AG6GR) [inline src/main/java/frc/robot/TeleopInput.java]: Cleanup comments

File: src/main/java/frc/robot/systems/FSMSystem.java [hunk 3]
```diff
@@ -1,132 +1,283 @@
...
-		}
-	}
+        switch (currentState) {
+            case RETRACTED:
+                if (input != null) {
+                    if(input.isShooterButtonPressed())
+                        return FSMState.SHOOT_RETRACTED;
+                    else if(input.isIntakeButtonPressed())
+                        return FSMState.INTAKING_RETRACTED;
+                    else if(input.isRampToggleButtonPressed() && canToggleRamp){
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/FSMSystem.java]: I assume this is to actuate once per button press? You might be able to use the getRawButtonPressed API to do this for you.
https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/GenericHID.html#getRawButtonPressed(int)

Comment (SiddharthKadari) [reply src/main/java/frc/robot/systems/FSMSystem.java]: Yes that is correct, just pushed a version that removes that redundancy

File: src/main/java/frc/robot/systems/FSMSystem.java (hunk 1 above)
Comment (SiddharthKadari) [reply src/main/java/frc/robot/systems/FSMSystem.java]: No I believe it was added when truing to test something, however it had no use in testing either. Just pushed a commit that removed redundant imports.

**Review by AG6GR — APPROVED** [review]

**Pull request merged**